packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -119,7 +119,10 @@ defineProperties(Element.prototype, {
     innerHTML: {
         get(this: Element): string {
             if (!featureFlags.ENABLE_ELEMENT_PATCH) {
-                if (isNodeShadowed(this) || isHostElement(this)) {
+                // If this element is in synthetic shadow, if it's a synthetic shadow host,
+                // or if any of its descendants are synthetic shadow hosts, then we can't
+                // use the native innerHTML because it would expose private node internals.
+                if (isNodeShadowed(this) || isHostElement(this) || containsHost(this)) {
```

Comment:
It looks like this expression can be extracted out of this function into a helper method.

Reply:
Done!